internal/descriptor/provider/provider_test.go

```diff
@@ -0,0 +1,44 @@
+package provider_test
+
+import (
+	"testing"
+
+	"github.com/open-component-model/ocm/pkg/contexts/ocm/compdesc"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	machineryruntime "k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/kyma-project/lifecycle-manager/api/v1beta2"
+	"github.com/kyma-project/lifecycle-manager/internal/descriptor/provider"
+)
+
+func TestCachedDescriptorProvider(t *testing.T) {
```

Comment:
I suggest breaking down this test suite into atomic test functions:

- Test that a newly created provider is empty
- Test `Add` method:
    - nil templates return an error
    - adding already existing entries returns no error
    - ...
 - Test `GetDescriptor` method:
    - ...